### ISSUE TITLE
Resuming from a mid-epoch checkpoint produces negative time estimates

### ISSUE THREAD

**Issue by fishbotics** [Lightning-AI/lightning#13124]
## 🐛 Bug

I have a checkpoint saved that I reloaded with `trainer(..., ckpt_path=...)` and then started training. The checkpoint was fairly early into the first epoch. Loading this epoch, I get some warnings and finally a super weird time estimate for training. The warnings are:

```
/usr/local/lib/python3.8/dist-packages/pytorch_lightning/callbacks/model_checkpoint.py:342: UserWarning: The dirpath has changed from '/workspace/mpinets_checkpoints/11dh7on3' to '/workspace/mpinets_checkpoints/<bound method DummyExperiment.nop of
 <pytorch_lightning.loggers.base.DummyExperiment object at 0x7fe38046ed60>>', therefore `best_model_score`, `kth_best_model_path`, `kth_value`, `last_model_path` and `best_k_models` won't be reloaded. Only `best_model_path` will be reloaded.
  warnings.warn(
```
and 
```
/usr/local/lib/python3.8/dist-packages/pytorch_lightning/loops/epoch/training_epoch_loop.py:135: UserWarning: You're resuming from a checkpoint that ended before the epoch ended. This can cause unreliable results if further training is done. Consider using an end-of-epoch checkpoint or enabling fault-tolerant training: https://pytorch-lightning.readthedocs.io/en/stable/advanced/fault_tolerant_training.html                       
```

And then finally, the training estimate comes back as negative. It looks like this :
```
Epoch 0:   0%|                                            | 11/63913 [05:54<-1:28:50, -34.17it/s, loss=0.00641, v_num=4rf6]
```

- PyTorch Lightning Version (e.g., 1.5.0): 1.6.3
- PyTorch Version (e.g., 1.10): 1.11
- Python version (e.g., 3.9): 3.8
- OS (e.g., Linux): Ubuntu 20.04
- CUDA/cuDNN version: 11.3
- GPU models and configuration: 8 A100s using ddp
- How you installed PyTorch (`conda`, `pip`, source): pip
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### Additional context

cc @tchaton @rohitgr7 @awaelchli

**Comment by awaelchli** [comment]
@fishbotics In your post here you included the warning message that tries to explain why this occurs:

/usr/local/lib/python3.8/dist-packages/pytorch_lightning/loops/epoch/training_epoch_loop.py:135: UserWarning: You're resuming from a checkpoint that ended before the epoch ended. This can cause unreliable results if further training is done. Consider using an end-of-epoch checkpoint or enabling fault-tolerant training: https://pytorch-lightning.readthedocs.io/en/stable/advanced/fault_tolerant_training.html                       

Resuming from a mid-epoch checkpoint is not supported in lightning unless you activate the experimental feature for fault-tolerant training. 

The progress bar acting out weird is probably related to that but we can look into it.

**Comment by fishbotics** [comment]
Thanks @awaelchli. All of my model checkpoints are saved mid-epoch. I use the ModelCheckpoint callback with a fixed time interval because my epochs take 35 hours. 

I sometimes want to pretrain my model on some data and either fine tune or just retrain on other data. What's the recommended workflow for that? If I try to load it as a checkpoint in the Trainer I get the aforementioned errors and timing issues. Should I directly load the model from the checkpoint and pass that into the trainer instead?

Likewise, I want to be able to train my model for several weeks but my cloud jobs timeout after 7 days. What would be the recommended workflow to restart a job with the same data? Is there a way to tell the ModelCheckpoint callback to save at the end of each epoch and also every hour? The docs say that the train_time_interval and every_n_epochs flags are mutually exclusive.

**Comment by fishbotics** [comment]
FWIW, I tried restarting the job with the fault tolerance flag and I still get negative time estimates.

**Comment by GGGGGGXY** [comment]
same issue here

**Comment by ibrahimmunir14** [comment]
I'm getting the same issue (negative time remaining and it/s), but when resuming from an end-of-epoch checkpoint. So I suspect the issue is not to do with saving mid-epoch.

**Comment by caillonantoine** [comment]
Same with end-of-epoch checkpoint !

**Comment by GGGGGGXY** [comment]
in pytorch_lightning/callbacks/progress/tqdm_progress.py line 269 
self.main_progress_bar.reset(convert_inf(total_batches))
make tqmd start from zero 


rewrite to:
self.main_progress_bar.total = convert_inf(total_batches)

fix this problem

**Comment by bryanmooremd** [comment]
I am also seeing this issue with end-of-epoch checkpoint saving.

**Comment by ductai199x** [comment]
@awaelchli I am noticing that resuming from an end-of-epoch checkpoint save also erase the tensorboard logs for ALL of the previous iterations. Is it possible to implement some test cases for progress bars and tensorboard logging so that future releases can be more stable? Because this are now a deal-breaker for me as a researcher.

**Comment by bryanmooremd** [comment]
I am still seeing very negative it/s when resuming training from a saved checkpoint, but it seems to have no correlation to actual training time per epoch, so for now the negative it/s seems like it can just be ignored:

Epoch 32:   7%|███████▉                                   |2353/32400[02:27<-1:55:37,-114.15it/s, loss=9.73e-07, v_num=lsdk]

**Comment by heiwang1997** [comment]
I'm using `wandb` and also have this issue. The `trainer/global_step` metric logged always restarts from 0 when I resume the training :(
I checked the actual `self.trainer.global_step` from within the model and that looks normal. Just wondering when this bug could be fixed.

**Comment by ductai199x** [comment]
> I'm using `wandb` and also have this issue. Also, the `trainer/global_step` metric logged always restarts from 0 when I resume the training :( I checked the actual `self.trainer.global_step` from within the model and that looks normal. Just wondering when this bug could be fixed.

Honestly I have no idea why this is not a priority. How can any researchers use Pytorch Lightning 1.6.x for any purposes, especially publishing papers? Now, PL is moving to version 1.7.x and this is still a big issue. Please fix this as soon as possible.

**Comment by awaelchli** [comment]
Fixed by #13962
@ductai199x Please give it a try. 

> Honestly I have no idea why this is not a priority. How can any researchers use Pytorch Lightning 1.6.x for any purposes, especially publishing papers? Now, PL is moving to version 1.7.x and this is still a big issue. Please fix this as soon as possible.

I respectfully disagree with your assessment here. This was a visual glitch and did not in any way affect the functionality of PyTorch Lightning in research. This should not have been preventing anyone from publishing papers. The progress bar values are estimates based on a running value, and should never be used as a way to benchmark the performance of software. I agree the way it was displayed was ugly and I'm glad we were able to fix it.

**Comment by bryanmooremd** [comment]
Just for further info, I was also using `wandb` when seeing this. I agree with @awaelchli that it seemed to be a visual glitch as the negative it/s was not correlated at all with training time per epoch. Was this something on Weights&Bias end?

**Comment by awaelchli** [comment]
Hey @bryanmooremd 
What exactly did WandB log and where? I have used W&B many times but never noticed it can do that. Maybe it's a recent feature. If you give me some more info, perhaps in a new issue, I can try to verify.

**Comment by gauenk** [comment]
I observe this issues too and I am not using WandB

**Comment by cogito288** [comment]
Same here. I also don't use WanDB. To make sure use end-of-epoch checkpoint, I use second last checkpoint of previous training. 
Epoch 46:  91%|███████████               | 9074/10009 [57:09<-1:02:51, -0.27it/s, loss=-0.786, v_num=6]

**Comment by famura** [comment]
This is related to #11029 which is still unsolved for me
